python/ray/serve/tests/unit/test_deployment_state.py

```diff
@@ -2009,12 +2010,25 @@ def test_basic_autoscaling(mock_deployment_state_manager, target_capacity_direct
         == DeploymentStatusTrigger.CONFIG_UPDATE_COMPLETED
     )
 
-    for replica in ds._replicas.get():
-        dsm.record_autoscaling_metrics(
-            replica._actor.replica_id,
-            2 if target_capacity_direction == "up" else 0,
-            None,
+    req_per_replica = 2 if target_capacity_direction == "up" else 0
```

Comment:
Did not change existing logic, but if `RAY_SERVE_COLLECT_AUTOSCALING_METRICS_ON_HANDLE=1` we need to call a different method to register the autoscaling metrics (`record_handle_metrics` vs `record_autoscaling_metrics`).